k6/vai/filter_change_configmaps.js

```diff
@@ -0,0 +1,397 @@
+const duration = (__ENV.DURATION || '2h')
+const diagnosticsInterval = (__ENV.DIAGNOSTICS_INTERVAL || "20m"); // # time unit
+// 2 requests per iteration (for change() func), so iteration rate is 1/2 of request rate
+const changeIPS = (__ENV.TARGET_RPS || 10) / 2
+const kubeconfig = k8s.kubeconfig(__ENV.KUBECONFIG, __ENV.CONTEXT)
+
+// Metrics
+const diagnosticsBeforeGauge = new Gauge('diagnostics_before_churn');
+const diagnosticsDuringTrend = new Trend('diagnostics_during_churn');
+const diagnosticsAfterGauge = new Gauge('diagnostics_after_churn');
+const churnOpsCounter = new Counter('churn_operations_total');
+
+const useFilters = (__ENV.USE_FILTERS !== undefined && __ENV.USE_FILTERS !== null) ? __ENV.USE_FILTERS === "true" ? true : false : true
+let defaultFilters = [
+  "sort=metadata.name&",
+  "filter=metadata.namespace!=p-4vgxn&",
+  "filter=metadata.namespace!=p-5xk4x&",
+  "filter=metadata.namespace!=cattle-fleet-clusters-system&",
+  "filter=metadata.namespace!=cattle-fleet-local-system&",
+  "filter=metadata.namespace!=cattle-fleet-system&",
+  "filter=metadata.namespace!=cattle-global-data&",
+  "filter=metadata.namespace!=cattle-impersonation-system&",
+  "filter=metadata.namespace!=cattle-provisioning-capi-system&",
+  "filter=metadata.namespace!=cattle-system&",
+  "filter=metadata.namespace!=cattle-ui-plugin-system&",
+  "filter=metadata.namespace!=fleet-default&",
+  "filter=metadata.namespace!=fleet-local&",
+  "filter=metadata.namespace!=kube-node-lease&",
+  "filter=metadata.namespace!=kube-public&",
+  "filter=metadata.namespace!=kube-system&"
+]
+
+export const options = {
+  insecureSkipTLSVerify: true,
+  tlsAuth: [
+    {
+      cert: kubeconfig["cert"],
+      key: kubeconfig["key"],
+    },
+  ],
+
+  summaryTrendStats: ['avg', 'min', 'med', 'max', 'p(95)', 'p(99)', 'count'],
+  setupTimeout: "3m",
+
+  scenarios: {
+    preChurnDiagnostics: {
+      executor: 'shared-iterations',
+      exec: 'preChurnDiagnostics',
+      vus: 1,
+      iterations: 1,
+      startTime: '0s',
+      maxDuration: '2m',
+      tags: { phase: 'pre-churn' }
+    },
+    change: {
+      executor: 'constant-arrival-rate',
+      exec: 'change',
+      preAllocatedVUs: changeVUs,
+      duration: duration,
+      rate: changeIPS,
+      startTime: '2m',
+      tags: { phase: 'churn' },
+    },
+    list: {
+      executor: 'per-vu-iterations',
+      exec: 'list',
+      vus: listVUs,
+      iterations: 9999, // setting this to a # we should never reach in order to ensure `list` runs until `change` scenario completes
+      maxDuration: duration,
+      startTime: '2m',
+      tags: { phase: 'list' },
+    },
+    duringChurnDiagnostics: {
+      executor: 'constant-arrival-rate',
+      exec: 'duringChurnDiagnostics',
+      preAllocatedVUs: 1,
+      rate: 1,
+      timeUnit: diagnosticsInterval,
+      duration: duration,
+      startTime: '2m',
+      tags: { phase: 'during-churn' },
+      thresholds: {
```

Comment:
Lets remove/move this last one to resolve `json: unknown field "thresholds"` errors

Reply:
Whoops, good catch. Fixed